app/src/main/java/com/github/fribourgsdp/radio/UserProfileActivity.kt

```diff
@@ -36,31 +50,45 @@ class UserProfileActivity : AppCompatActivity(), PlaylistAdapter.OnPlaylistClick
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_user_profile)
 
-        firebaseAuth = FirebaseAuth.getInstance()
-        checkUser()
+        try {
+            user = User.load(this)
+        } catch (e: java.io.FileNotFoundException) {
+            //this should never happen as a user is created and saved in the first activity
+            createDefaultUser()
+        }
```

Comment:
If you are sure this will never happen, I think you can remove the try catch. Or throw in the catch an UnexpectedException or something like that.

Reply:
My thought was that even though it should never happen, it could for reason we don't control. Better be safe than sorry.